sub-packages/bionemo-llm/src/bionemo/llm/lightning.py

```diff
@@ -258,6 +259,8 @@ def __init__(
         self._data_step = data_step
         self._forward_step = forward_step
         self.model_transform = model_transform
+        self.train_ppl = torchmetrics.text.Perplexity(ignore_index=-100)
+        self.valid_ppl = torchmetrics.text.Perplexity(ignore_index=-100)
```

Comment:
https://lightning.ai/docs/torchmetrics/stable/pages/overview.html#metrics-in-distributed-data-parallel-ddp-mode

Reply:
https://lightning.ai/docs/torchmetrics/stable/pages/overview.html#metric-kwargs

Reply:
Any changes to the defaults we should make there? I used torchmetrics with the HF pretraining and it seemed to work well, so I don't think the oversampling is too big of an issue

Reply:
`sync_on_compute` is on by default so we are good on that end. I wonder if we need to specify `process_group` to skip non-last-pp-stage devices.

Reply:
I also don't have a clear idea on how to treat tp devices.